indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3501,7 +3527,11 @@ private Map<PartitionIdType, OrderedSequenceNumber<SequenceOffsetType>> generate
       } else {
         // if we don't have a startingOffset (first run or we had some previous failures and reset the sequences) then
         // get the sequence from metadata storage (if available) or Kafka/Kinesis (otherwise)
-        OrderedSequenceNumber<SequenceOffsetType> offsetFromStorage = getOffsetFromStorageForPartition(partitionId);
+        final Map<PartitionIdType, SequenceOffsetType> metadataOffsets = getOffsetsFromMetadataStorage();
```

Comment:
Is this just a style change? Or did you mean to fetch the offsets outside the loop?

Reply:
The offsets for all the partitions are fetched outside the loop and this map is used by `getOffsetFromStorageForPartition`

Reply:
Fixed